src/Features/CSharp/Portable/UseExpressionBodyForLambda/UseExpressionBodyForLambdaHelpers.cs

```diff
@@ -0,0 +1,227 @@
+            return (canOffer, fixesError: false);
+        }
+
+        public static LambdaExpressionSyntax Update(
+            SemanticModel semanticModel, LambdaExpressionSyntax declaration, bool useExpressionBody)
+        {
+            return UpdateWorker(semanticModel, declaration, useExpressionBody).WithAdditionalAnnotations(Formatter.Annotation);
+        }
+
+        private static LambdaExpressionSyntax UpdateWorker(
+            SemanticModel semanticModel, LambdaExpressionSyntax declaration, bool useExpressionBody)
+        {
+            return useExpressionBody 
+                ? WithExpressionBody(declaration)
+                : WithBlockBody(semanticModel, declaration);
+        }
+
+        private static LambdaExpressionSyntax WithExpressionBody(LambdaExpressionSyntax declaration)
+        {
+            if (!TryConvertToExpressionBody(
+                    declaration, declaration.SyntaxTree.Options, ExpressionBodyPreference.WhenPossible,
+                    out var expressionBody, out _))
+            {
+                return declaration;
+            }
+
+            var updatedDecl = declaration.WithBody(expressionBody);
+
+            // If there will only be whitespace between the arrow and the body, then replace that
+            // with a single space so that the lambda doesn't have superfluous newlines in it.
+            if (declaration.ArrowToken.TrailingTrivia.All(t => t.IsWhitespaceOrEndOfLine()) &&
+                expressionBody.GetLeadingTrivia().All(t => t.IsWhitespaceOrEndOfLine()))
+            {
+                updatedDecl = updatedDecl.WithArrowToken(updatedDecl.ArrowToken.WithTrailingTrivia(SyntaxFactory.ElasticSpace));
+            }
+
+            return updatedDecl;
+        }
+
+        private static LambdaExpressionSyntax WithBlockBody(
+            SemanticModel semanticModel, LambdaExpressionSyntax declaration)
+        {
+            var expressionBody = GetExpressionBody(declaration);
+            var createReturnStatementForExpression = CreateReturnStatementForExpression(semanticModel, declaration);
+
+            if (!expressionBody.TryConvertToStatement(
+                    semicolonTokenOpt: default,
+                    createReturnStatementForExpression,
+                    out var statement))
+            {
+
+                return declaration;
+            }
+
+            // If the user is converting to a block, it's likely they intend to add multiple
+            // statements to it.  So make a multi-line block so that things are formatted properly
+            // for them to do so.
+            return declaration.WithBody(SyntaxFactory.Block(
+                SyntaxFactory.Token(SyntaxKind.OpenBraceToken).WithAppendedTrailingTrivia(SyntaxFactory.ElasticCarriageReturnLineFeed),
+                SyntaxFactory.SingletonList(statement),
+                SyntaxFactory.Token(SyntaxKind.CloseBraceToken)));
+        }
+
+        private static bool CreateReturnStatementForExpression(
```

Comment:
>CreateReturnStatementForExpression [](start = 28, length = 34)

This logic doesn't seem to account for task builders (custom task types). For instance, `ValueTask` and `ValueTask<T>`.
I'm not sure whether we have helpers to deal with that yet.

Reply:
ValueTask will be hte interesting one.  But we don't reference that type yet.  We should create a helper somewhere to ask if something is TaskLike.

Reply:
Did you get a chance to look at this? (I didn't see changes related to this)
Could you file a follow-up issue for `ValueTask` and other task-like scenarios?

Reply:
So i'm of two minds here.  

1. I have no problem creating  follow-up issue.  
2. However, here's the honest state of the world.  Those followup issues *never* get looked at.  They simply bloat up the repo.  So, i'm willing to do that if you really want me to (just ask again).  However, i would prefer the default policy not be to just open bugs if there's a honest belief that they will actually serve no purpose.  :)

Reply:
That's indeed likely. Fine to let it go.